Description: Fix missing check for hidraw and support DINAH CM119B gpio PTT configuration.
Author: Alan Crosswell <n2ygk@weca.org>
Bug-Debian: https://bugs.debian.org/1014017

1. Fixes configure.ac to look for <linux/hidraw.h>
2. Extends the test for the controller product ID.

diff --git soundmodem-0.20.orig/configure.ac soundmodem-0.20/configure.ac
index b9ccd87..a54600d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -52,7 +52,7 @@ AC_CHECK_LIB(posix4,sched_setscheduler)
 AC_CHECK_LIB(uuid,GUID_NULL)
 AC_CHECK_FUNCS(getopt_long,,[AC_LIBOBJ([getopt]) AC_LIBOBJ([getopt1])])
 AC_CHECK_HEADERS(sys/audioio.h stropts.h sys/conf.h sys/soundcard.h sys/ioctl.h time.h inttypes.h net/if_arp.h)
-AC_CHECK_HEADERS(pty.h getopt.h syslog.h sched.h linux/sockios.h sys/ioccom.h linux/ppdev.h)
+AC_CHECK_HEADERS(pty.h getopt.h syslog.h sched.h linux/sockios.h sys/ioccom.h linux/ppdev.h linux/hidraw.h)
 AC_CHECK_HEADERS(sys/socket.h linux/if.h,,,
 [[
 #if HAVE_SYS_SOCKET_H
diff --git soundmodem-0.20.orig/soundcard/ptt.c soundmodem-0.20/soundcard/ptt.c
index 1b168e1..377d374 100644
--- a/soundcard/ptt.c
+++ b/soundcard/ptt.c
@@ -271,7 +271,7 @@ int pttinit(struct pttio *state, const char *params[])
 		  (
 		    (hiddevinfo.vendor == 0x0d8c	// CM108/109/119
 			&& hiddevinfo.product >= 0x0008
-			&& hiddevinfo.product <= 0x000f
+			&& hiddevinfo.product <= 0x0012
 		    )
 		    ||
 		    (hiddevinfo.vendor == 0x0c76 &&	// SSS1621/23
